ruby/generate-map/index.rb

```diff
@@ -0,0 +1,59 @@
+require 'open-street-map'
+require 'json'
+require 'uri'
+require 'net/http'
+require 'base64'
+
+def error(res, message)
+  return res.json({"success": false, "message": message})
+end
+
+def generate_tile_number(lat_as_deg, lon_as_deg, zoom)
+  lat_as_rad = lat_as_deg/180 * Math::PI
+  n = 2.0 ** zoom
+  x = ((lon_as_deg + 180.0) / 360.0 * n).to_i
+  y = ((1.0 - Math::log(Math::tan(lat_as_rad) + (1 / Math::cos(lat_as_rad))) / Math::PI) / 2.0 * n).to_i
+  return {:x => x, :y =>y}
+end
+
+def main(req, res)
+  lng = nil
+  lat = nil
+  begin
+    payload = JSON.parse(req.payload)
+    lng = payload["lng"]
+    lat = payload["lat"]
+  rescue
+    return error(res, "The payload must contain lng and lat")
+  end
+
+  if (lng == nil || lat == nil)
+    return error(res, "The payload must contain lng and lat")
+  elsif (!lng.is_a?(Integer) && !lat.is_a?(Integer)) || (!lng.is_a?(Float) && !lat.is_a?(Float))
+    return error(res, "The longitude and latitude must be a number")
+  elsif (!(-180..180).include? lng)
+    return error(res, "The longitude must be between -180 and 180")
+  elsif (!(-90..90).include? lat)
+    return error(res, "The latitude must be between -90 and 90")
+  end
+  # Check if User-Agent env var exists
+  user_agent_header = nil
+  begin
+    user_agent_header = req.env['USER_AGENT']
+  rescue
+    return error(res, "Missing 'User-Agent' enviornment variable.")
```

Comment:
Let's make this match the variable:

```suggestion
    return error(res, "Missing 'USER_AGENT' function variable.")
```